### PR TITLE
feat: add crictl tool and ensure lxcfs parent dir in manager

### PR DESCRIPTION
**Pull request by yxxhero**
This pull request introduces several changes aimed at improving functionality, enhancing validation, and refining dependencies for the project. The most important changes include adding support for `crictl` in the Docker image, implementing a utility function to validate the `lxcfs` parent directory, and updating dependency versions in `go.mod`.

### Dockerfile enhancements:
* Added `crictl` support by downloading and extracting the `crictl` binary during the Docker build process and copying it to `/usr/local/bin` for use in the container. (`Dockerfile.agent`, [[1]](diffhunk://#diff-7a3de9cd336efea6d1d2594cf86e83112c95390648c1e1ba9790bcc33fc5a0e3R30-R33) [[2]](diffhunk://#diff-7a3de9cd336efea6d1d2594cf86e83112c95390648c1e1ba9790bcc33fc5a0e3R49)

### Validation improvements:
* Introduced a new utility function `EnsureLxcfsParentDir` in `pkg/utils/utils.go` to validate that the parent directory of the `lxcfs` path is `lxc`. Added comprehensive unit tests to ensure correct behavior across various edge cases. (`pkg/utils/utils.go`, [[1]](diffhunk://#diff-81ddbadfb415ccbb9c7af84f11668d1aa5e53c34025bf86d4702f16b4e42f045R1-R15); `pkg/utils/utils_test.go`, [[2]](diffhunk://#diff-dbb00af8f58edf5a904cecaf0c5eb4c07979ce8711d22f826bfd70ce5e9634c9R1-R63)
* Integrated the `EnsureLxcfsParentDir` function into the `cmd/manager/main.go` file to validate the `lxcfsPath` during startup. (`cmd/manager/main.go`, [cmd/manager/main.goR78-R82](diffhunk://#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633R78-R82))

### Dependency updates:
* Updated `k8s.io/utils` and `github.com/emicklei/go-restful/v3` to newer versions in `go.mod` to incorporate bug fixes and improvements. (`go.mod`, [[1]](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R9-R17) [[2]](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L54)

### Code refinements:
* Replaced `strings.TrimSuffix` with `strings.TrimRight` in `pkg/admission/mutate.go` for improved handling of trailing slashes in paths. (`pkg/admission/mutate.go`, [[1]](diffhunk://#diff-4548b0e9793e2fcaadc308567ac4a72feb84798c39d054a7b8aeccfe69f35dbdL122-R149) [[2]](diffhunk://#diff-4548b0e9793e2fcaadc308567ac4a72feb84798c39d054a7b8aeccfe69f35dbdL195-R204)
* Refactored mount propagation logic in `ensureVolumeMount` to use a dedicated mapping function, simplifying the code and improving readability. (`pkg/admission/mutate.go`, [pkg/admission/mutate.goL122-R149](diffhunk://#diff-4548b0e9793e2fcaadc308567ac4a72feb84798c39d054a7b8aeccfe69f35dbdL122-R149))

## Summary by CodeRabbit

* **新功能**
  * 镜像中新增集成 crictl 工具，便于容器管理。
  * 新增路径校验机制，确保 lxcfs 路径的父目录为 "lxc"。

* **修复**
  * 优化了挂载路径的斜杠处理，提升路径兼容性。
  * 挂载点的传播模式根据名称动态分配，提高挂载安全性。

* **依赖升级**
  * 升级了部分依赖库，提升兼容性和稳定性。

* **测试**
  * 新增对路径校验工具函数的单元测试，增强代码可靠性。

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+func EnsureLxcfsParentDir(path string) error {
+	lxcDir := filepath.Dir(strings.TrimRight(path, "/"))
+	if !strings.HasSuffix(lxcDir, "lxc") {
```
Comment (felipecrs) [inline pkg/utils/utils.go:11]: I normally use `/var/lib/lxcfs-on-kubernetes` to avoid any potential conflict with lxcfs running in the node by other means like LXC.

Comment (yxxhero) [reply pkg/utils/utils.go:11]: @felipecrs yeah. It's a good idea
